frontend/src/utilities/notebookControllerUtils.ts

```diff
@@ -172,3 +174,16 @@ export const generatePvc = (pvcName: string, pvcSize: string): PersistentVolumeC
     phase: 'Pending',
   },
 });
+
+export const checkNotebookRunning = (notebook?: Notebook): boolean =>
+  !!(
+    notebook?.status?.readyReplicas &&
+    notebook?.status?.readyReplicas >= 1 &&
+    notebook?.metadata.annotations?.['opendatahub.io/link']
+  );
+
+export const getUserStateFromDashboardConfig = (
+  translatedUsername: string,
+  notebookControllerState?: NotebookControllerUserState[],
```

Comment:
Any reason you don't push this optionality issue up to the caller? It seems odd to make a stand alone method have a key yet optional variable 🤔

Reply:
The `notebookControllerState` could be undefined if there is no user in it, it could probably not be created at all.

Reply:
As I mentioned before, you need to start thinking of separation of concerns. This method, as it is written, has absolutely no logical sense to have `notebookControllerState` be optional. You're making it optional because the caller might have an optional chunk of data -- because you're thinking "I know where this data comes from". 

Push that concern up to the caller, this should be a required prop; they can do `|| []` if they need to. Your method NEEDS this array to work, why on earth would it declare a required property for it to work as optional? That makes no sense from the perspective of this method.

Reply:
That makes sense, I will update it.